### PR TITLE
fix(FOROME-1269): input-numeric with react-number-format

### PR DESCRIPTION
**Pull request by EvgeniySobolev**
[Task](https://quantori.atlassian.net/browse/FOROME-1269)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -26,6 +26,7 @@
     "react-indiana-drag-scroll": "2.1.0",
     "react-intl": "5.15.8",
     "react-modal": "3.13.1",
+    "react-number-format": "^4.9.3",
```
Comment (AlMaQntr) [inline package.json]: Please, fix version number

**Review by AlMaQntr — APPROVED** [review]

File: src/ui/input-numeric/input-numeric.tsx [hunk 2]
```diff
@@ -79,12 +89,17 @@ export const InputNumeric = ({
...
+        <NumberFormat
           className={cn(styles.inputNumeric__input)}
           type="text"
           value={displayValue}
           disabled={disabled}
-          onChange={e => changeValue(e.target.value)}
+          thousandSeparator={' '}
+          onValueChange={(v: any) => {
+            if (v.floatValue !== inputValue) {
+              changeValue(v.floatValue || 0)
```
Comment (AlMaQntr) [inline src/ui/input-numeric/input-numeric.tsx]: empty input should return `null` value

**Review by QSdmitrioul — APPROVED** [review]

**Comment by AndreyUstyumenko** [comment]
1. No styles change when focus
2. Tabindex is lost. Focus on the first input and then hit Tab to switch to the second one. Focus is removed from the first input but not set on the second one.
Screen Shot 2022-07-04 at 17 54 38

File: src/ui/input-numeric/input-numeric.tsx [hunk 3]
```diff
@@ -79,12 +95,20 @@ export const InputNumeric = ({
...
       <div className={cn(styles.inputNumeric__inputSection)}>
-        <input
+        <NumberFormat
           className={cn(styles.inputNumeric__input)}
           type="text"
           value={displayValue}
           disabled={disabled}
-          onChange={e => changeValue(e.target.value)}
+          thousandSeparator={' '}
+          onValueChange={(v: any) => {
```
Comment (AndreyUstyumenko) [inline src/ui/input-numeric/input-numeric.tsx]: don't shorten variables. V must be `value`. Also, remove `any` and pick a proper type. I assume it's `number` apparently.

File: src/ui/input-numeric/input-numeric.tsx [hunk 4]
```diff
@@ -14,45 +15,60 @@ interface IInputNumericProps {
...
+  const displayValue = value === '' ? value : inputValue
+  let enteredNumber: number | null = inputValue ?? null
 
   useEffect(() => {
     setInputValue(+value)
   }, [value])
 
-  function changeValue(newValue: number | string) {
+  function checkValue(value: number): number {
+    const isFloatNumber = Boolean(isFloat)
```
Comment (AndreyUstyumenko) [inline src/ui/input-numeric/input-numeric.tsx]: useless variable. Use just `!!isFloat` later on.

**Review by AndreyUstyumenko — APPROVED** [review]

**Pull request merged**